### PR TITLE
Generate warnings for code examples

### LINKED ISSUE #1190: Generate warnings in docs before C code examples

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #1190.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Oh that's crazy, my clippy lint is now showing up! :D

File: src/codegen/doc/format.rs [hunk 1]
```diff
@@ -71,7 +71,10 @@ fn code_blocks_transformation(
 fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
     if let (_, Some(after)) = try_split(entry, LANGUAGE_SEP_BEGIN) {
         if let (before, Some(after)) = try_split(after, LANGUAGE_SEP_END) {
-            out.push_str(&format!("\n```{}", before));
+            out.push_str(&format!(
+                "\n\n**⚠️ The following code is in {0} ⚠️**\n\n```{0}",
```
Comment (bilelmoussaoui) [inline src/codegen/doc/format.rs:75]: I don't think it's needed here. Because often those might be just plain types or functions. Worth trying it locally

Comment (GuillaumeGomez) [reply src/codegen/doc/format.rs:75]: I'll add a check for the language itself, but even so, still needs to be marked as such. ;)

Comment (bilelmoussaoui) [reply src/codegen/doc/format.rs]: But those can be sometimes replaced by the docs links no?

Comment (GuillaumeGomez) [reply src/codegen/doc/format.rs]: Wait I'm lost now. In which case a \`\`\` code block can be replaced by a link?

File: src/codegen/doc/format.rs [hunk 2]
```diff
@@ -71,7 +71,10 @@ fn code_blocks_transformation(
 fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
     if let (_, Some(after)) = try_split(entry, LANGUAGE_SEP_BEGIN) {
         if let (before, Some(after)) = try_split(after, LANGUAGE_SEP_END) {
-            out.push_str(&format!("\n```{}", before));
+            if before != "text" && before != "rust" {
```
Comment (sdroege) [inline src/codegen/doc/format.rs:74]:
```suggestion
            if !["text", "rust"].contains(before) {
```

File: src/codegen/doc/format.rs [hunk 3]
```diff
@@ -104,6 +107,9 @@ fn format(
                 ret.push_str(&replace_symbols(before, env, in_type));
 
                 let (lang, after) = get_markdown_language(after);
+                if lang != "text" {
```
Comment (sdroege) [inline src/codegen/doc/format.rs:110]: Why not check for Rust here?

File: src/codegen/doc/format.rs (hunk 1 above)
Comment (sdroege) [reply src/codegen/doc/format.rs]: I guess when it's a one-line example?

Can you create a gtk-rs PR with this? Then we can check the generated docs :)

Comment (GuillaumeGomez) [reply src/codegen/doc/format.rs]: We don't have the docs in the different repositories, so it wouldn't change anything.

Comment (sdroege) [reply src/codegen/doc/format.rs]: It's built as part of the github pages

Comment (GuillaumeGomez) [reply src/codegen/doc/format.rs]: It would require to merge the PR for that. :)

Comment (sdroege) [reply src/codegen/doc/format.rs]: Are you sure? It should be available under https://guillaumegomez.github.io/gtk-rs-core/ and similar.

Comment (GuillaumeGomez) [reply src/codegen/doc/format.rs]: Not sure about that. :-/

**Comment by GuillaumeGomez** [comment]
Updated!

**Review by sdroege — APPROVED** [review]

**Comment by SafariMonkey** [comment]
@GuillaumeGomez I just looked at this change and it looks like the negation on the language check was dropped? I'm guessing as part of applying [this](https://github.com/gtk-rs/gir/pull/1191#discussion_r654891668) change. Currently it appears to me that it only shows the warning before `text` or `rust` blocks, which I'm pretty sure is the opposite of the intention.

**Comment by GuillaumeGomez** [comment]
From the diff, I don't see how it's possible. :-/

**Comment by sdroege** [comment]
> From the diff, I don't see how it's possible. :-/

Because you forgot the `!` :)

**Comment by GuillaumeGomez** [comment]
Oh crap...

**Pull request merged**